### ISSUE TITLE
If file name contain  a " . "  termimal do not works well

### ISSUE THREAD

**Issue by webMing** [rust-lang/rust-analyzer#13234]
### Des: if rust filename contain a "." , terminal don`t show any error message  or alters.

```rust
c.c.rs  //terminal don`t show any alters

cc.rs //termimal works well and you can see any error message 

```
### Env
vscode
Version: 1.71.1 (Universal)
Commit: e7f30e38c5a4efafeec8ad52861eb772a9ee4dfb
Date: 2022-09-08T19:41:32.136Z (6 days ago)
Electron: 19.0.12
Chromium: 102.0.5005.167
Node.js: 16.14.2
V8: 10.2.154.15-electron.0
OS: Darwin x64 21.1.0
Sandboxed: No

rustc 1.63.0 (4b91a6ea7 2022-08-08)

rust-analyzer last updated  9/12/2022, 17:30:54

**Comment by bjorn3** [comment]
What do you exactly mean? As far as I know rust-analyzer doesn't have anything terminal related.

**Comment by webMing** [comment]
> What do you exactly mean? As far as I know rust-analyzer doesn't have anything terminal related.
I  have update my question  more readable

**Comment by webMing** [comment]
> What do you exactly mean? As far as I know rust-analyzer doesn't have anything terminal related.

Your means this issue caused by vscode ?

**Comment by bjorn3** [comment]
What is the issue you are having?

**Comment by webMing** [comment]
> 

sorry . i update issue again.

**Comment by webMing** [comment]
> What is the issue you are having?

I have change my issue . please have a look again.

**Comment by bjorn3** [comment]
Thanks. I now understand what the issue is you are having. This is probably a problem with rust-analyzer. I can't try to reproduce it right now. Maybe someone else can take a look?

**Comment by lnicola** [comment]
Some features might not work very well outside of a Cargo project. Once you rename the file, it's outside of the project tree. For example, not even `cargo check` will look at it.

**Comment by webMing** [comment]
> Some features might not work very well outside of a Cargo project. Once you rename the file, it's outside of the project tree. For example, not even `cargo check` will look at it. That even applies if you try to make it a binary:
> 
> ```
> $ mkdir examples
> $ touch examples/c.c.rs
> $ cargo run --example c.c
>    Compiling hello v0.1.0 (/home/grayshade/Projects/hello)
> error: invalid character `.` in crate name: `c.c`
> 
> error: could not compile `hello` due to previous error
> ```

```shell
$ cargo new test_file_name
     Created binary (application) `test_file_name` package
$ cd test_file_name/src
$ touch c.c.rs
$ cd ../../
$ code test_file_name
```

In c.c.rs file  there is no any lint message show.
```rust 
fn foo()-> i32 {

}

struct Obj {
    str:&str,
}

```


In main.rs file shows that : "   mismatched types expected `i32`, found `()`"

```rust
fn abc() ->i32 {
    
}

```

**Comment by lnicola** [comment]
Yes, that's what I'm saying. Your file is not part of the project.

**Comment by lnicola** [comment]
To add `x.rs` to the project you need to write `mod x;` in the parent module. Your file can't be added like this.

**Comment by webMing** [comment]
> To add `x.rs` to the project you need to write `mod x;` in the parent module. Your file can't be added like this.
> 
> The problem is not the dot, but the fact that your file is not declared as a module. The dot just prevents you from declaring it. You can rename your file to `c_c.rs`, add the module declaration and it will work.
> 
> In your example, `cargo check` will not show any errors either.


All find that : if i move c.c.rs to src/bin and add main method it also not work.  last i get this : if rust project module tree not contain  the module any file in this module will not show lint message if something wrong. oh i`m not good at english.

**Comment by bjorn3** [comment]
> Also find that : if i move c.c.rs to src/bin and add main method to it that also not work.

That doesn't even compile:

```
error: invalid character `.` in crate name: `c.c`

error: could not compile `foo` due to previous error
```

**Comment by ChayimFriedman2** [comment]
As said this is likely an error with your setup.
